src/createReducer.js

```diff
@@ -303,7 +303,7 @@ function createReducer<M, L>(structure: Structure<M, L>) {
     ) {
       let result = state
       const initial = getIn(result, `initial.${field}`)
-      if (initial === undefined && payload === '') {
+      if (payload === '') {
```

Comment:
to not make this a breaking change for thoose who use `''` to reset value use bellow suggestion
```suggestion
      if ((initial === undefined && payload === '') || payload === undefined) {
```

Reply:
this will break only 2 tests, fix them to allow `undefined` value
```
  ● reducer.immutable › reducer.immutable.change › should allow setting an initialized field to ''
  ● reducer.immutable › reducer.immutable.change › should not modify a value if called with undefined
```

Reply:
this should be `request change`